doc/cprover-manual/loop-invariant-synthesizer.md

```diff
@@ -0,0 +1,28 @@
+[CPROVER Manual TOC](../)
+
+## Loop Invariant Synthesis with goto-instrument
+
+This is a loop invariant synthesizer built in goto-instrument that 
+synthesize [invariant clause](contracts-invariants.md) for a 
+pre-existing GOTO binary, annotate the invariant clause into 
+the GOTO binary, and produce the annotated binary as output.
+
+Loop invariant synthesized by the synthesizer must be inductive 
+in the sense that 
+1. the instrumented loop invariant checks success, no matter other 
+assertions fail or success;
+2. the base-case is from the whole GOTO program with a main 
+function as the entry point.
+
+For different synthesis strategy, there are different prerequisites
+on the input GOTO binary, and different guarantees on the synthesized
+loop invariants, summarized in the
+following table. 
```

Comment:
Why do we have this table? Are there other strategies? If so, why is the flag for "enumerative" strategy called "--synthesize-loop-invariants"? Shouldn't it be called something like `--synthesize-loop-invariants-enumerative"?

Reply:
There are no other strategies now. But I expect at least we will have a CHC-based synthesizer as another strategy. 

Yes, I agree that the flag should be more specified when there are more than one strategies. However, I think it would be better to keep is simple now.

Reply:
I am not sure if the CHC-based synthesizer also aims to plug the invariant back into GOTO, and whether it would follow the same workflow as the enumerative one.

I agree on the simplicity part, but users might expect `--synthesize-loop-invariants` to run a portfolio of all available synthesizers. To avoid ambiguity and churn in future, I would suggest renaming the flag to be more specific and explicit.

Reply:
I see. I modified the text and also the name of the flag in code.